standalone figures (pdf + png) and beamer slides

ptoche opened the pull request:
I have reproduced all the book figures. They come in standalone pdf and png format. Other formats can be created from the 'knitr' source as desired. A set of beamer slides with each figure (plus a small number of extra ones) is created by knitr. An example of presentation based on Thomas Piketty's 26 November 2014 Sao Paulo  powerpoint is also included. In the process I rewrote the data import, as I found I needed to understand where the data had come from in the spreadsheets (not because I did it better, I didn't).  I have saved the dataframes for convenience. For instance, Figure 1.1 is produced by the code chunk labeled Figure_1_1 with the dataframe named df_1_1 and is saved as Figure_1_1.pdf and Figure_1_1.png. 

The README.md file lists the file in the repository named 'Slides'. There are some issues with the dual-axis plots, which I intend to fix shortly. I noticed that some figures were missing from the master repository, so my code may be useful in filling the gaps. I have not covered the tables. All the figures were generated with ggplot, though I intend to produce some rCharts too for some selected figures (the ones that have more than 5 series on them). Comments welcome, naturally!

stedy commented:
I have been looking over these changes, quite a lot of work! 

Some thoughts:

I think its fine to have all this new code in a separate directory but perhaps we should consider making the entire Slides folder as a separate branch since the original idea was to have all the code in R markdown.

One of the original intentions was to have people generate their own pdf/png files locally, and not have any pdf/png files on github that can't be reproduced locally.

I made some changes to dataFiguresTables.rnw because I could not get it to compile without errors, my changes might be a bit hacky since I was just trying to get the pdf file to compile.

ptoche commented:
Thanks Zach!

1. no objection to separating the Slides, but on the other hand it may be useful to have all the R Piketty stuff in one place, no? I understand that knitr and rmarkdown are separate, but most R people would be open-minded to both. The con for joining them is that my repo is a bit heavy with all the figures and slides.

My intention in creating the pdf/png files and having them in the repo is that people who do not use R or who cannot be bothered to recreate the figures, could just download them. I thought Thomas Piketty himself might be interested in the colored versions to spruce up his powerpoints.

2. sorry dataFiguresTables.rnw did not compile for you. I'll look into it, it's possible I made some local edits without realizing that they affected elsewhere (I do usually check, but I did finish rather tired).

3. A few hours before finishing the slides, I discovered that Quandl has all the Piketty data! It might have been smarter to write code to download from there, rather than read from the complicated xlsx...

Thanks for your feedback and support Zach!

ptoche commented:
follow up on 2. There were problems in the code of chapter 5, which I have fixed. This is the story: when I started processing the worksheets, I was focused on getting the data for the specific figures of the book, but at some point I thought it might be useful to grab more data to produce more figures, as that thought crossed my mind I thought it would be easier to find my way around the data if I saved the worksheets pretty much 'as is' (not in long form) with names like 'TS5.1.Rda' for worksheet 'TS5.1' and save the subsetted data frames in long forms for each figure with names like 'df_5_1.Rda'. I also figured it would be easier if all the figure data frames came in the Year/variable/value format. With that in mind, I modified parts of the code of chapter 5, one of the earliest chapters I worked on, but I did not make the changes everywhere and obviously forgot to check that it all compiled at the end. All this late in the night. I hope it's fixed now.  Thanks a lot for the feedback. It's a lot easier to make fixes now than it will be in a few weeks time.  ;-)
